src/dtx/dtx_common.c

```diff
@@ -791,6 +798,9 @@ dtx_begin(struct ds_cont_child *cont, struct dtx_id *dti,
 
 	if (daos_is_zero_dti(dti)) {
 		daos_dti_gen(&dth->dth_xid, true);
+		rc = vos_dtx_rsrvd_init(dth);
+		/** should never be non-zero in this case */
```

Comment:
I wasn’t quite sure here.   The issue is that if rsrvd_fini is called, the assertion I added for the list being empty will trigger.  If it’s not called, this is not needed

Reply:
For zero_dti case, if we call vos_dtx_rsrvd_init() when dtx_begin(), then we need vos_dtx_rsrvd_fini() when dtx_end(). If some assertion is triggered at that time, then there should be something else to be fixed, otherwise we will leak some DRAM.